### PR TITLE
unittests covering 1005 of resource.py

### PR DESCRIPTION
**Pull request by zancas**
@pjbreaux 

Start by running:

`py.test --cov-report=term-missing --cov=./resource.py -v -s ./test/test_resource.py`

in the appropriate location.

Then inspect the code in test/test_resource.py

Fixes: https://github.com/F5Networks/f5-common-python/issues/201

tests are in test/test_resource.py

### CODE REVIEW

File: f5/bigip/test/test_resource.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+def test_Resource__local_update():
+    r = Resource(mock.MagicMock())
+    stash = r._meta_data.copy()
+    r._local_update({'test': 1})
+    assert stash == r._meta_data
+    r.__dict__.pop('_meta_data')
+    assert r.__dict__ == {'test': 1}
+
+
+def test_Resource__check_keys_valid_rict():
```
Comment (pjbreaux) [inline f5/bigip/test/test_resource.py]: rdict for func name

File: f5/bigip/test/test_resource.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        assert r.selfLink == u".../~Common~test_create"
+
+
+def test__activate_URI():
+    r = Resource(mock.MagicMock())
+    r._meta_data['allowed_lazy_attributes'] = []
+    r._meta_data['attribute_registry'] = {u"tm:": u"SPAM"}
+    r._meta_data['bigip']._meta_data = {'hostname': 'TESTDOMAIN'}
+    TURI = 'https://localhost/mgmt/tm/'\
+           'ltm/nat/~Common~testnat/?ver=11.5&a=b#FOO'
```
Comment (pjbreaux) [inline f5/bigip/test/test_resource.py]: Test uri with no query args, but a hash frag.

File: f5/bigip/test/test_resource.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+        r.delete(force=False)
+        assert r.__dict__ == {'deleted': True}
+
+    def test_invalid_force(self):
+        r = Resource(mock.MagicMock())
+        r._meta_data['uri'] = 'URI'
+        mock_session = mock.MagicMock()
+        r._meta_data['bigip']._meta_data = {'icr_session': mock_session}
+        with pytest.raises(InvalidForceType) as IFTEIO:
+            r.delete(force='true')
```
Comment (pjbreaux) [inline f5/bigip/test/test_resource.py]: Is it the case that force is always a string? And could it be force='True' as well?

File: f5/bigip/test/test_resource.py [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+                 {'reference': {'link': 'https://...'}}]
+        attrs = {'get.return_value':
+                 MockResponse({u"generation": 0,
+                               u"items": items})}
+        mock_session = mock.MagicMock(**attrs)
+        c._meta_data['bigip']._meta_data =\
+            {'icr_session': mock_session,
+             'hostname': 'TESTDOMAINNAME'}
+        c.generation = 0
+        c.get_collection()
```
Comment (pjbreaux) [inline f5/bigip/test/test_resource.py]: assert collection contains the items you expect?

File: f5/bigip/test/test_resource.py (hunk 3 above)
Comment (zancas) [inline f5/bigip/test/test_resource.py]: If "force" is ever anything besides a Python bool (i.e. `True` or `False`) then an exception is raised.  That's the purpose of this test.

File: f5/bigip/test/test_resource.py (hunk 4 above)
Comment (zancas) [inline f5/bigip/test/test_resource.py]: I'll add an assertion.

**Comment by zancas** [comment]
I'm going to merge this since @pjbreaux may not be available over the weekend, and @swormke might want to do packaging.

**Pull request merged**